internal/packages/archetype/package.go

```diff
@@ -0,0 +1,121 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package archetype
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/formatter"
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/packages"
+)
+
+// PackageDescriptor defines configurable properties of the package archetype
+type PackageDescriptor struct {
+	Manifest packages.PackageManifest
+}
+
+// CreatePackage function bootstraps the new package based on the provided descriptor.
+func CreatePackage(packageDescriptor PackageDescriptor) error {
+	baseDir := packageDescriptor.Manifest.Name
+	_, err := os.Stat(baseDir)
+	if err == nil {
+		return fmt.Errorf(`package "%s" already exists`, baseDir)
+	}
+
+	logger.Debugf("Write package manifest")
+	err = renderResourceFile(packageManifestTemplate, &packageDescriptor, filepath.Join(baseDir, "manifest.yml"))
+	if err != nil {
+		return errors.Wrap(err, "can't render package manifest")
+	}
+
+	logger.Debugf("Write package changelog")
+	err = renderResourceFile(packageChangelogTemplate, &packageDescriptor, filepath.Join(baseDir, "changelog.yml"))
+	if err != nil {
+		return errors.Wrap(err, "can't render package changelog")
+	}
+
+	logger.Debugf("Write docs readme")
+	err = renderResourceFile(packageDocsReadme, &packageDescriptor, filepath.Join(baseDir, "docs", "README.md"))
```

Comment:
Does it make sense to write the `docs/README.md` file or should we write the `_dev/build/docs/README.md` file instead... or both?

Reply:
Hm... it would mean that there is a template without any placeholders, which in this case will the same as outputted one. Not sure if it doesn't introduce confusion here.

Reply:
I see. Okay, lets leave this as-is. In any case the template file is not strictly required according to the spec but the (generated) README.md file is. So I think what you have here is good for the purposes of getting started with a new package.